backend.pm

```diff
@@ -528,7 +528,11 @@ sub ssh_start
 		sleep(1);
 	}
 
-	my $sshcmd = 'ssh ' . ($key ? "-i $key " : " ") . "$user\@$host";
+	my $sshcmd = 'ssh ';
+	$sshcmd .= ($key ? "-i $key " : " ");
+	$sshcmd .= $self->{'ssh_opts'} . ' ';
+	$sshcmd .= "$user\@$host";
+	$sshcmd =~ s/'/\\'/g;
```

Comment:
I'm not sure that we want to escape everything like this here.

What exactly is this needed for?

Reply:
It is because of the next line, where we have a `-c '$sshcmd'` and if we do not escape, then we would brake the quotes there.
And I guess it is coming from the `ssh_opts` as these could contain something like `-o RekeyLimit='1G 1h'` or some other value which might need a escape...

But in general, I think this is wrong, it should be:
```suggestion
	$sshcmd =~ s/'/'"'"'/g;
```

Looking at the ssh-client options, there are not many which might need it, so maybe we could just skip it?

Reply:
@cfconrad i run some tests and it doesnt look like effect us if we remove it. So i did so for simplicity. We can added later if it is needed.

Reply:
cool

I would say this example would makes trouble:
```
perl ... "--backend=ssh:user=azureuser:key_file=/root/.ssh/id_rsa:host=51.144.169.227:ssh_opts=-o RekeyLimit='1G 2h'" ;

Reply:
i modified this line to include the potential threat $cmd .= ':ssh_opts=\'-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o RekeyLimit=\'1G 2h\'\''; and it works. 
So i dont think it would be a problem to skip the escape here. wdyt?

Reply:
it turns out @cfconrad was right(http://aquarius.suse.cz/tests/5287#step/run_ltp/125) using a small hack so i reverted the line.

Reply:
also this is a good example why I would escape these single quotes:

```
clemix@linux-7k5u:~> export FOO=BLUB
clemix@linux-7k5u:~> script -f -c 'echo "$FOO"'
Script started, file is typescript
BLUB
Script done, file is typescript
clemix@linux-7k5u:~> script -f -c 'echo '"'"'$FOO'"'"''
Script started, file is typescript
$FOO
Script done, file is typescript
clemix@linux-7k5u:~>